apps/src/gamelab/GameLabP5.js

```diff
@@ -435,6 +646,7 @@ GameLabP5.prototype.getCustomMarshalBlockedProperties = function () {
 
 GameLabP5.prototype.getCustomMarshalObjectList = function () {
   return [
+    { instance: GameLabGame },
```

Comment:
Totally fine if that's not viable right now.  I'm just wishing for a
situation where we have our own GameLabSprite, GameLabGroup, etc that we
can extend so easily.

On Tue, Mar 29, 2016 at 1:54 PM, Chris Pirich <notifications@github.com>
wrote:

> In apps/src/gamelab/GameLabP5.js
> <https://github.com/code-dot-org/code-dot-org/pull/7528#discussion_r57798797>
> :
>
> > @@ -435,6 +646,7 @@ GameLabP5.prototype.getCustomMarshalBlockedProperties = function () {
> >
> >  GameLabP5.prototype.getCustomMarshalObjectList = function () {
> >    return [
> > +    { instance: GameLabGame },
>
> Maybe. The tricky thing about p5play is that they don't typically use
> prototype at all. It would be easier to extend if they did. And probably
> more efficient since we wouldn't have to patch each object as it is
> instantiated.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/7528/files/6282551cbc590eb2122233267d0316f5ccf571aa#r57798797>
>



-- 
Brad Buchanan
Software Engineer, Code.org
brad@code.org